rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -142,8 +142,10 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
 
   void on_participant_discovery(
     eprosima::fastdds::dds::DomainParticipant *,
-    eprosima::fastrtps::rtps::ParticipantDiscoveryInfo && info) override
+    eprosima::fastrtps::rtps::ParticipantDiscoveryInfo && info,
+    bool & should_be_ignored) override
   {
+    should_be_ignored = false;
```

Comment:
These changes will only build with Fast DDS 2.10.0 onwards, so I suggest changing the version here:

https://github.com/ros2/rmw_fastrtps/blob/cf9bf8323e34956f98e925cc8c1501e63ccf8eb5/rmw_fastrtps_shared_cpp/CMakeLists.txt#L49-L50